Cargo.toml

```diff
@@ -9,14 +9,43 @@ repository = "https://github.com/gfx-rs/naga"
 keywords = ["shader", "SPIR-V"]
 license = "MPL-2.0"
 
+[lib]
+crate-type = ["cdylib", "rlib"]
+
 [dependencies]
 bitflags = "1"
 fxhash = "0.2"
 log = "0.4"
 num-traits = "0.2"
 spirv = { package = "spirv_headers", version = "1" }
 
+wasm-bindgen = "0.2"
```

Comment:
this should be wasm-target-only, see [wgpu-rs](https://github.com/gfx-rs/wgpu-rs/blob/4b16959e3e5e35178ba7b22cf0ad182344615da6/Cargo.toml#L79) as an example

Reply:
I briefly tried making wasm-bindgen optional ... but couldn't quite make it work. Maybe a question of the right cmd line incantation

Reply:
This is now flagged by target_arch

Reply:
This is better, but unfortunately it's still a subject to the problem with cargo-vendor, which will happily consider these dependencies to be needed for Gecko.
To make it clear, I think we should keep the wasm-naga as a separate project for now.

Reply:
agreed